File: viz/prepare_truth_data.py
import pandas as pd
from epiweeks import Week, Year

df1 = pd.read_csv('data-truth/JHU/truth_JHU-Incident Deaths.csv')
df1.rename(columns={'value': 'inc_death'}, inplace=True)

df2 = pd.read_csv('data-truth/JHU/truth_JHU-Incident Cases.csv')
df2.rename(columns={'value': 'inc_case'}, inplace=True)

# merge cases and deaths into one dataframe
df = df1.merge(df2, on=['date', 'location', 'location_name'])

# add epi weeks for aggregation
df.date = pd.to_datetime(df.date)
df['epi_week'] = df.date.apply(lambda x: Week.fromdate(x).week)
df['epi_year'] = df.date.apply(lambda x: Week.fromdate(x).year)

# aggregate to weekly incidence
df = df.groupby(['location', 'location_name', 'epi_year', 'epi_week']).aggregate(
    {'date': max, 'inc_death': sum, 'inc_case':sum}).reset_index()

# only keep Saturdays
df = df[df.date.dt.day_name() == 'Saturday']

# reformat
df = df[['date', 'location', 'location_name', 'inc_case', 'inc_death']].sort_values(['date', 'location'])

# export
df.to_csv('truth_to_plot.csv', index=False)

File: viz/prepare_truth_data.R
library(dplyr)
library(here)
library(readr)
library(purrr)
library(tidyr)

# FIXME: find a way to get this information directly from the config file
# without hardcoding target types
var_files <- c(
  "inc_death" = "ECDC/truth_ECDC-Incident Deaths.csv",
  "inc_case" = "ECDC/truth_ECDC-Incident Cases.csv",
  "inc_hosp" = "OWID/truncated_OWID-Incident Hospitalizations.csv"
)
truth <- var_files |>
  imap(~ read_csv(here("data-truth", .x)) %>% mutate(name = .y)) |>
  bind_rows()

# get legacty data
legacy <- c("ECDC/final/", "OWID/final") |>
  map(~ tail(list.files(here("data-truth", .x), full.names = TRUE), 1)) |>
  map(~ read_csv(.x, show_col_types = FALSE)) |>
  bind_rows() |>
  mutate(name = gsub(" ", "_", target_variable)) |>
  filter(name %in% names(var_files)) |>
  anti_join(truth, by = c("location_name", "location", "date", "name"))

truth <- truth |>
  bind_rows(legacy) |>
  # add epi weeks for aggregation
  mutate(date = lubridate::ymd(date),
         epi_week = lubridate::epiweek(date),
         epi_year = lubridate::epiyear(date)) |>
  group_by(location, epi_year, epi_week, name) |>
  # aggregate to weekly incidence
  summarise(date = max(date),
            value = sum(value),
            .groups = "drop") |>
  # only keep Saturdays
  pivot_wider() |>
  filter(lubridate::wday(date, label = TRUE) == "Sat") |>
  # reformat
  arrange(date, location) |>
  select(-epi_week, -epi_year) |>
  as.data.frame()

write_csv(truth, "viz/truth_to_plot.csv", quote = "needed")
save(truth, file = "viz/truth.RData")
